### ISSUE TITLE
VS 2017: LOG doesn't compile in template function

### ISSUE THREAD

**Issue by andy-bis** [SergiusTheBest/plog#120]
Hi! The issue is very similar to #114 and #116, but currently this is compilation error.
The error is the following
> error C2760: syntax error: unexpected token 'this', expected 'id-expression'

The code for reproduce
```
#include <plog/Log.h>
#include <iostream>

template <class T>
void f(T t)
{
	LOGE << t;

	// comment LOGE line and uncomment next to find out root reason
	/*__if_exists(this)
	{
		std::cout << "exists";
	}
	__if_not_exists(this)
	{
		std::cout << "not exists";
	}*/
}

int main()
{
	f(0);
	return 0;
}
```
The reason of error is __if_exists and __if_not_exists statements used inside PLOG_GET_THIS. It seems, support of this statement inside templates is broken for some versions of VS 2017.
We have found that:
* _MSC_VER == 1910 it compiles
* _MSC_VER == 1912 and _MSC_VER == 1916 - it does not compile. I assume that it will not compile for all intermediate versions

So, possibly, condition for PLOG_GET_THIS should be updated to something like this
`(_MSC_VER >= 1600 && _MSC_VER < 1912 || _MSC_VER > 1916)`

Or even disable 'this' capture feature by default, maybe? It is not used for most formatters.

**Comment by SergiusTheBest** [comment]
Hi @andy-bis,

It's strange, my VS2017 has `_MSC_VER == 1912` but compiles your sample fine. The full version string is: `Microsoft Visual Studio Community 2017, Version 15.5.6`. Could you say yours?

> Or even disable 'this' capture feature by default, maybe? It is not used for most formatters.

It would be a breaking change, so I'd better avoid that.

**Comment by SergiusTheBest** [comment]
However if turn on conformance mode (/permissive-) 
![image](https://user-images.githubusercontent.com/4660722/52472870-15f8d680-2b9d-11e9-9069-502632857820.png)
then it will produce an error:
```
error C2760: syntax error: unexpected token 'this', expected 'id-expression'
```
Is this the error you observe?

**Comment by andy-bis** [comment]
@SergiusTheBest,
> error C2760: syntax error: unexpected token 'this', expected 'id-expression'

Yes, this is exactly the error I observe.

Currently, I'm using VS2017 that has _MSC_VER == 1916. Full version is "Microsoft Visual Studio Professional 2017 , Version 15.9.9"

Now I see that issue is not related to _MSC_VER, but it is related to enabled 'Conformance mode' option.
I assume that error occurs for _MSC_VER >= 1912 because 'Conformance mode' option is introduced on-by-default in VS 2017, version 15.5: https://devblogs.microsoft.com/cppblog/msvc-conformance-improvements-in-visual-studio-2017-version-15-5/

Сurrently in our project, we have used workaround that disables PLOG_GET_THIS() for _MSC_VER >= 1912. But this workaround is wrong, considering that we want to move to CSV formatter in the future. So any other ideas how to make it compatible with 'Conformance mode' will be appreciated 😄

**Comment by SergiusTheBest** [comment]
Marking this as fixed by #132.
